gamification/src/main/java/com/appcoins/wallet/gamification/repository/BdsPromotionsRepository.kt

```diff
@@ -86,29 +125,40 @@ class BdsPromotionsRepository(private val api: GamificationApi,
     return ForecastBonus(ForecastBonus.Status.INACTIVE)
   }
 
-  override fun getGamificationStats(wallet: String): Single<GamificationStats> {
+  override fun getGamificationStats(wallet: String): Observable<GamificationStats> {
     return getUserStats(wallet)
-        .map { mapToGamificationStats(it) }
+        .map {
+          mapToGamificationStats(it)
+        }
         .flatMap {
           local.setGamificationLevel(it.level)
               .toSingle { it }
+              .toObservable()
         }
   }
 
-  private fun map(status: Status): GamificationStats {
+  override fun getGamificationLevel(wallet: String): Single<Int> {
+    return getGamificationStats(wallet).map { it.level }
+        .filter { it >= 0 }
+        .lastOrError()
+        .onErrorReturn { -1 }
+  }
```

Comment:
I believe the filter here doesn't fit very well, since you want to return a single. For my understanding, you want to just use the value from the API, so a flag should be used as recommended above and then use a toSingleOrError. If the intention is to return the db in case the API returns an error then it's good, but maybe leave a comment if that is it.